Sources/do-not-disturb/DoNotDisturb.swift

```diff
@@ -12,22 +12,35 @@ public struct DoNotDisturb {
 		DistributedNotificationCenter.default().postNotificationName(NSNotification.Name("com.apple.notificationcenterui.dndprefs_changed"), object: nil, deliverImmediately: true)
 	}
 
-	private static func enable() {
-		set("dndStart", value: 0 as CFPropertyList)
-		set("dndEnd", value: 1440 as CFPropertyList)
-		set("doNotDisturb", value: true as CFPropertyList)
-		commitChanges()
+	private static func restartNotificationCenter() {
+		NSRunningApplication.runningApplications(withBundleIdentifier: "com.apple.notificationcenterui").first?.forceTerminate()
+	}
 
-		// For some reason `doNotDisturb` does not take the first time around
-		// TODO: Figure out why
-		sleep(for: 0.4)
-		set("doNotDisturb", value: true as CFPropertyList)
+	private static func enable() {
+		if !isEnabled {
+			set("doNotDisturb", value: true as CFPropertyList)
+			set("doNotDisturbDate", value: Date() as CFPropertyList)
+			commitChanges()
+			restartNotificationCenter()
+		}
 	}
 
 	private static func disable() {
+		if isEnabled {
+			set("doNotDisturb", value: false as CFPropertyList)
+			set("doNotDisturbDate", value: nil)
+			commitChanges()
+			restartNotificationCenter()
+			restoreMenubarIcon()
+		}
+	}
+
+	private static func restoreMenubarIcon() {
```

Comment:
For some reason the Menubar Icon only listens to the `dndStart` and `dndEnd` when it is disabled. So setting them and unsetting them re-enables it, but it requires the sleep in-between the two. I tried reducing it and 0.3 was the lowest it would work for, for me. Feel free to test around and see if there's a different workaround

Reply:
Maybe try the trick mention in https://github.com/sindresorhus/do-not-disturb/pull/3?